### PR TITLE
NF: Remove `else ->` when possible

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
#17352 duplicated. For some reason I can't re-open it

When `when` is used on an enum, the `else ->` can always be transformed into a complete enumeration of the enum's entries.

This ensure that each time a new entry is added, there is a warning and the developer must decide whether or not a change is needed. While the `else` may hide some required change.

So, for the sake of future developement, I removed `else ->` when it makes sense.

I should note that I kept some `else ->` when the `when` consider only a very small part of all possible enum value, in order to reduce the size of the code.

### CODE REVIEW

**Review by BrayanDSO — APPROVED** [review]
LGTM

**Review by mikehardy — CHANGES_REQUESTED** [review]
I like the else-to-specific-cases transform, but I counted 5 other changes in here in 2 categories of nullability changes and code deletion changes. Those may be okay on their own as well but they don't seem like they're intended to be included in the single commit given it only discuses the else transform?

**Comment by Arthur-Milchior** [comment]
Visibly wrong conflict resolution. reverted. Thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/MediaPlayer.kt [hunk 1]
```diff
@@ -243,6 +283,7 @@ class MediaPlayer :
         IDLE,
         INITIALIZED,
         PREPARING,
+
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/MediaPlayer.kt:286]: what's this?

```suggestion
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt [hunk 2]
```diff
@@ -85,12 +84,7 @@ data class SoundOrVideoTag(val filename: String) : AvTag() {
 }
 
 /** In python, this is a union of [TTSTag] and [SoundOrVideoTag] */
-open class AvTag
-
-/**
- * [Regex] used to identify the markers for sound files
- */
-val SOUND_RE = Pattern.compile("\\[sound:([^\\[\\]]*)]").toRegex()
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt:93]: not exactly an else removal :-)

**Review by mikehardy — CHANGES_REQUESTED** [review]
I'm ready to hit that merge button but still two tiny things

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/MediaPlayer.kt (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/MediaPlayer.kt:286]: Don't know how I introduced this one

File: AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt:93]: It was not used anymore and I cleaned this part of the code.
I can add part and we can remove it later

**Review by mikehardy — APPROVED** [review]
fixed the trivial compile error introduced in response to my request to not have deletions unrelated to the else transform, the import for the re-added chunk was missing

in the hands of CI now...

**Pull request merged**